cookiejar.js

```diff
@@ -67,11 +67,22 @@
         if (this instanceof Cookie) {
             var parts = str.split(";").filter(function (value) {
                     return !!value;
-                }),
-                pair = parts[0].match(/([^=]+)=([\s\S]*)/),
-                key = pair[1],
-                value = pair[2],
-                i;
+                });
+            var i;
+
+            var pair = parts[0].match(/([^=]+)=([\s\S]*)/);
+            if (!pair) {
+                console.warn("Invalid cookie header encountered. Header: '"+str+"'");
+                return;
+            }
+
+            var key = pair[1];
+            var value = pair[2];
+            if (!key || !value) {
```

Comment:
this should probably be:

```javascript
if ( typeof key === 'undefined' || typeof value === 'undefined' ) {
```

Otherwise, if you have a false-y value like 0 we'll blow up.

Reply:
I guess ```!'0'``` actually evaluates false. Is an empty string a valid value per the spec?

Reply:
> 2.  If the name-value-pair string lacks a %x3D ("=") character,
>       ignore the set-cookie-string entirely.
>
>   3.  The (possibly empty) name string consists of the characters up
>       to, but not including, the first %x3D ("=") character, and the
>       (possibly empty) value string consists of the characters after
>       the first %x3D ("=") character.
>
>   4.  Remove any leading or trailing WSP characters from the name
>       string and the value string.
>
>   5.  If the name string is empty, ignore the set-cookie-string
>       entirely.

So we should probably do:

```javascript
if ( typeof key !== 'string' || key.length === 0 || typeof value !== 'string' ) {
```

Reply:
Yeah. Empty strings are falsy. The string '0' is also falsy. Everything extracted via the regex would be a string, unless we were to try to cast it. https://developer.mozilla.org/en-US/docs/Glossary/Falsy